File: bubbleSort/bubbleSort.go
package main

import "fmt"

func bubbleSort(arr []int) []int {
	ln := len(arr)

	for i := 0; i < ln; i++ {
		for j := 0; j < ln-1; j++ {
			if arr[j] > arr[j+1] {
				temp := arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}
	return arr
}

func main() {
	arr := []int{1, 4, 7, 8, 2, 5}
	sortArr := bubbleSort(arr)
	for i := 0; i < len(arr); i++ {
		fmt.Printf("%d", sortArr[i])
	}
}

File: bubbleSort/bubbleSort.c
#include <stdio.h>

int *bubbleSort(int arr[], int size)
{
    for (int i = 0; i < size; i++)
    {
        for (int j = 0; j < size - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    return arr;
}
int main()
{
    int arr[] = {1, 5, 4, 9, 2, 8};
    int size = sizeof(arr) / sizeof(arr[0]);

    int *sortedArray = bubbleSort(arr, size);

    for (int i = 0; i < size; i++)
    {
        printf("%d ", sortedArray[i]);
    }

    return 0;
}
